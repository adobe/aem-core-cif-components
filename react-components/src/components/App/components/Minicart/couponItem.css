.root {
    display: grid;
    grid-gap: 0 1rem;
    grid-template-columns: 1fr min-content;
    margin: 0 1.5rem;
}
.couponName {
    line-height: 1.5rem;
}
