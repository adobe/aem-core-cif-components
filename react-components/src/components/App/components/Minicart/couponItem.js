/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { useCartState } from './cartContext';
import Kebab from './kebab';
import Section from './section';

import classes from './couponItem.css';

const CouponItem = () => {
    const [{ cart, removeCoupon }] = useCartState();

    const appliedCoupon = cart.applied_coupon ? cart.applied_coupon.code : null;

    return (
        <div className={classes.root}>
            <div className={classes.couponName}>
                Coupon <strong>{appliedCoupon}</strong> applied.
            </div>
            <Kebab>
                <Section text="Remove coupon" onClick={removeCoupon} icon="Trash" />
            </Kebab>
        </div>
    );
};

export default CouponItem;
