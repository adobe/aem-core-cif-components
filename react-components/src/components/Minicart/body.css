.root {
    display: grid;
    grid-template-rows: min-content min-content 1fr;
    overflow: auto;
}
