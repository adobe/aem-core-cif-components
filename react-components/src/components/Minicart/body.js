/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { useTranslation } from 'react-i18next';

import LoadingIndicator from '../LoadingIndicator';

import EmptyMinicartBody from './emptyMinicartBody';
import classes from './body.css';
import ProductList from './productList';
import CartOptions from './cartOptions';
import Error from './error';
import CouponForm from './couponForm';
import CouponItem from './couponItem';

import { useCartState } from './cartContext';
import DiscountList from './discountList';

const CartLoadingIndicator = () => {
    const [t] = useTranslation('cart');

    return <LoadingIndicator>{t('cart:fetching-data', 'Fetching cart data...')}</LoadingIndicator>;
};

const Body = () => {
    const [{ isEditing, cart, isLoading, errorMessage }] = useCartState();
    const isEmpty = cart && Object.entries(cart).length > 0 ? cart.items.length === 0 : true;

    if (isLoading) {
        return <CartLoadingIndicator />;
    }

    if (errorMessage) {
        return <Error />;
    }

    if (isEmpty) {
        return <EmptyMinicartBody />;
    }

    if (isEditing) {
        return <CartOptions />;
    }

    const cartItems = cart.items;
    const couponFragment = !cart.applied_coupon ? <CouponForm /> : <CouponItem />;
    const discountFragment =
        cart.prices.discounts && cart.prices.discounts.length > 0 ? (
            <DiscountList discounts={cart.prices.discounts} />
        ) : (
            <div />
        );

    return (
        <div className={classes.root}>
            <ProductList cartItems={cartItems} />
            {discountFragment}
            {couponFragment}
        </div>
    );
};

export default Body;
