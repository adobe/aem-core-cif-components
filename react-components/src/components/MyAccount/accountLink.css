/*******************************************************************************
 *
 *    Copyright 2021 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
.root {
    border-bottom: 1px solid rgb(var(--venia-border));
    width: 100%;
}

.content {
    align-items: center;
    display: grid;
    gap: 0.5rem;
    grid-auto-flow: column;
    grid-template-columns: 2rem 1fr;
    height: 3rem;
    justify-items: start;
}

.icon {
    align-items: center;
    color: rgb(var(--venia-teal));
    display: flex;
    justify-content: center;
    justify-self: center;
}

.text {
    font-size: 0.875rem;
    line-height: 1rem;
}

.root_normalPriority {
    composes: root;
}
