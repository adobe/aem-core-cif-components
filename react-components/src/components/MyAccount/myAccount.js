/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { LogOut as SignOutIcon } from 'react-feather';

import AccountLink from './accountLink';
import LoadingIndicator from '../LoadingIndicator';

import classes from './myAccount.css';
import { useUserContext } from '../../context/UserContext';
import { func } from 'prop-types';

const SIGN_OUT = 'Sign Out';

const MyAccount = props => {
    const { showMenu } = props;
    const [{ currentUser, isSignedIn, inProgress }, { signOut }] = useUserContext();

    if (inProgress) {
        return (
            <div className={classes.modal_active}>
                <LoadingIndicator>{'Signing In'}</LoadingIndicator>
            </div>
        );
    }

    if (!isSignedIn) {
        showMenu();
    }

    return (
        <div className={classes.root}>
            <div className={classes.user}>
                <h2 className={classes.title}>{`${currentUser.firstname} ${currentUser.lastname}`}</h2>
                <span className={classes.subtitle}>{currentUser.email}</span>
            </div>
            <div className={classes.actions}>
                <AccountLink onClick={signOut}>
                    <SignOutIcon size={18} />
                    {SIGN_OUT}
                </AccountLink>
            </div>
        </div>
    );
};

MyAccount.propTypes = {
    showMenu: func.isRequired
};

export default MyAccount;
