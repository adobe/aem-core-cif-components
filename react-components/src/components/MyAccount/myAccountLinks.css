.links {
    display: grid;
    gap: 1rem;
    padding: 1.5rem;
}

.links button {
    border-bottom: 0px;
}
