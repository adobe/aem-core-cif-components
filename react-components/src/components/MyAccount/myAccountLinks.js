/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { Book as BookIcon, Info as InfoIcon } from 'react-feather';
import { useTranslation } from 'react-i18next';

import AccountLink from './accountLink';
import SignOutLink from './signOutLink';

import classes from './myAccountLinks.css';

import { func } from 'prop-types';

const MyAccountLinks = props => {
    const { showAddressBook, showAccountInformation } = props;
    const [t] = useTranslation('account');

    return (
        <div className={classes.links}>
            <AccountLink onClick={showAddressBook}>
                <BookIcon size={18} />
                {t('account:address-book', 'Address Book')}
            </AccountLink>
            <AccountLink onClick={showAccountInformation}>
                <InfoIcon size={18} />
                {t('account:address-information', 'Address Information')}
            </AccountLink>
            <SignOutLink />
        </div>
    );
};

MyAccountLinks.propTypes = {
    showAddressBook: func.isRequired,
    showAccountInformation: func.isRequired
};

export default MyAccountLinks;
