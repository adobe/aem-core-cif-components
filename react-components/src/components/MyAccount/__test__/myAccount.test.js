/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/

import React from 'react';
import { fireEvent, wait } from '@testing-library/react';
import { render } from 'test-utils';
import { CartProvider } from '../../Minicart';
import MyAccount from '../myAccount';

// avoid console errors logged during testing
console.error = jest.fn();

describe('<MyAccount>', () => {
    it('renders the component', async () => {
        const { asFragment } = render(
            <CartProvider initialState={{ cartId: null }} reducerFactory={() => state => state}>
                <MyAccount showMenu={jest.fn()} showAccountInformation={jest.fn()} showChangePassword={jest.fn()} />
            </CartProvider>
        );
        await wait(() => {
            expect(asFragment()).toMatchSnapshot();
        });
    });

    it('renders the loading indicator when inProgress is true', async () => {
        const stateWithInProgress = { inProgress: true };

        const { asFragment } = render(
            <CartProvider initialState={{ cartId: null }} reducerFactory={() => state => state}>
                <MyAccount showMenu={jest.fn()} showAccountInformation={jest.fn()} showChangePassword={jest.fn()} />
            </CartProvider>,
            { userContext: stateWithInProgress }
        );
        await wait(() => {
            expect(asFragment()).toMatchSnapshot();
        });
    });

    it('call the callback function when sign out button is clicked', async () => {
        const mockReducerFactory = jest.fn(state => state);

        const { getByText } = render(
            <CartProvider initialState={{ cartId: null }} reducerFactory={() => mockReducerFactory}>
                <MyAccount showMenu={jest.fn()} showAccountInformation={jest.fn()} showChangePassword={jest.fn()} />
            </CartProvider>
        );

        await wait(() => {
            expect(getByText('Sign Out')).not.toBeUndefined();
            fireEvent.click(getByText('Sign Out'));

            expect(mockReducerFactory).toHaveBeenCalledTimes(1);
        });
    });
});
