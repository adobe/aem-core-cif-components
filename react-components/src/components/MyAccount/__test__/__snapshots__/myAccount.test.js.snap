// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyAccount> renders the component 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="user"
    >
      <h2
        class="title"
      >
         
      </h2>
      <span
        class="subtitle"
      />
    </div>
    <div
      class="actions"
    >
      <button
        classes="[object Object]"
        type="button"
      >
        <span>
          <span>
            <svg
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="11"
                rx="2"
                ry="2"
                width="18"
                x="3"
                y="11"
              />
              <path
                d="M7 11V7a5 5 0 0 1 10 0v4"
              />
            </svg>
          </span>
          <span>
            Change Password
          </span>
        </span>
      </button>
      <button
        classes="[object Object]"
        type="button"
      >
        <span>
          <span>
            <svg
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 19.5A2.5 2.5 0 0 1 6.5 17H20"
              />
              <path
                d="M6.5 2H20v20H6.5A2.5 2.5 0 0 1 4 19.5v-15A2.5 2.5 0 0 1 6.5 2z"
              />
            </svg>
          </span>
          <span>
            Address Book
          </span>
        </span>
      </button>
      <button
        classes="[object Object]"
        type="button"
      >
        <span>
          <span>
            <svg
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <line
                x1="12"
                x2="12"
                y1="16"
                y2="12"
              />
              <line
                x1="12"
                x2="12.01"
                y1="8"
                y2="8"
              />
            </svg>
          </span>
          <span>
            Account Information
          </span>
        </span>
      </button>
      <button
        classes="[object Object]"
        type="button"
      >
        <span>
          <span>
            <svg
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
              />
              <polyline
                points="16 17 21 12 16 7"
              />
              <line
                x1="21"
                x2="9"
                y1="12"
                y2="12"
              />
            </svg>
          </span>
          <span>
            Sign Out
          </span>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<MyAccount> renders the loading indicator when inProgress is true 1`] = `
<DocumentFragment>
  <div
    class="modal_active"
  >
    <div
      class="root"
    >
      <span
        class="indicator loader-img"
      />
      <span
        class="message"
      >
        Signing In
      </span>
    </div>
  </div>
</DocumentFragment>
`;
